usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
 
 
 (function() {
+    var INTERVAL_FREQUENCY_MS = 60000;
+    var EXPIRY_BUFFER = 300000;
+    var HOURS_PER_DAY = 24;
+    var NOON = 12;
+    var CHANNEL = "HiFi.Google.Calendar";
+    var GREEN_TEXT = [168, 255, 168];
+    var RED_TEXT = [255, 168, 168];
+    var GREEN = [125, 255, 125];
+    var RED = [255, 125, 125];
+
     var that;
 
     this.remotelyCallable = [
+        "refreshToken",
         "enteredMeetingZone",
-        "leftMeetingZone",
-        "updateSignColor",
-        "showNoScheduledEvents",
-        "addEvent",
-        "clearEventList"
+        "leftMeetingZone"
     ];
 
 
     this.preload = function(entityID) {
         that = this;
         that.entityID = entityID;
-        that.entityProperties = Entities.getEntityProperties(that.entityID, ['id', 'name', 'type', 'parentID']);
-        that.room = {};
-        if (that.entityProperties.name.indexOf("_SCHEDULE") === -1 && 
-            that.entityProperties.name.indexOf("_OCCUPANTS") === -1 && 
-            that.entityProperties.type === "Text") {
+        that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+        if (that.entityProperties.userData.length !== 0) {
+            try {
+                that.userData = JSON.parse(that.entityProperties.userData);
+            } catch (e) {
+                console.log(e, "Could not parse userData");
+                return;
+            }
+            that.roomScheduleID = that.userData.roomScheduleID; 
+            that.roomOccupantListID = that.userData.roomOccupantListID; 
+        } else {
+            console.log("Please enter appropriate userData to enable functionality of this server script.");
```

Comment:
in this error case, it doesn't look like you're stopping the functions below from executing. if that's true, please add such protection

Reply:
Isn't calling "return" sufficient protection?  We don't want the script to run if the userData is incomplete or incorrect.